controller/src/main/java/io/pravega/controller/store/stream/InMemoryStreamMetadataStore.java

```diff
@@ -104,18 +104,27 @@ Scope newScope(final String scopeName) {
 
     @Override
     @Synchronized
-    public CompletableFuture<Boolean> updateConfiguration(final String scopeName,
-                                                          final String streamName,
-                                                          final StreamConfiguration configuration,
-                                                          final OperationContext context,
-                                                          final Executor executor) {
+    public CompletableFuture<List<Integer>> updateConfiguration(final String scopeName,
+                                                                final String streamName,
+                                                                final StreamConfiguration configuration,
+                                                                final OperationContext context,
+                                                                final Executor executor) {
         if (scopes.containsKey(scopeName)) {
             String scopeStreamName = scopedStreamName(scopeName, streamName);
             if (!streams.containsKey(scopeStreamName)) {
                 return FutureHelpers.
                         failedFuture(StoreException.create(StoreException.Type.DATA_NOT_FOUND, scopeStreamName));
             }
-            return streams.get(scopeStreamName).updateConfiguration(configuration);
+            Stream stream = streams.get(scopeStreamName);
+            return stream.getActiveSegments().thenCompose(segments -> {
```

Comment:
since state fencing is done in stream.updateConfiguration where it sets the state to UPDATING,
this check here is problematic.. it should be moved inside streams if it has successfully set the state to UPDATING.. 
Otherwise what can happen here is if a sealing is ongoing, it may getActiveSegments as old segments here and think it can go ahead with update.. and in update, by the time it tries to set State to UPDATING, sealing may have completed and stream state may have been set back to ACTIVE (unless you have reverted and retained the SEALED state). 
So in such a case update will not honor the contract that we cant update a sealed stream